### PR TITLE
Migrate GraphQL API to TS

### PR DESCRIPTION
**Pull request by phildarnowsky-broad**
Most of the changes fall into two categories:

* Changing from CommonJS-style `require/module.exports` exports to ES6-style `import/export`.
* Adding types, almost all of which are just `any` at this point.

We also now emit CommonJS module code, which both node and the browser can deal with, rather than ES6. In the future, it might be worth configuring things so that the API uses CommonJS and the browser uses ES6.

### CODE REVIEW

File: graphql-api/src/lodash.d.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// There's a @types package for lodash but adding it as a dependency
```
Comment (mattsolo1) [inline graphql-api/src/lodash.d.ts:1]: Strange

**Review by mattsolo1 — APPROVED** [review]
Looks great to me! I poked around through various datasets in the online demo and saw no issues.

File: graphql-api/src/lodash.d.ts (hunk 1 above)
Comment (phildarnowsky-broad) [reply graphql-api/src/lodash.d.ts:1]: could very well be, will try

Comment (phildarnowsky-broad) [reply graphql-api/src/lodash.d.ts]: @mattsolo1 ...except that even though it passes CI, now trying to build an image for deployment crashes. Putting this back like it was.

**Pull request merged**